warehouse/ingest-core/src/main/java/datawave/ingest/config/TableConfigCache.java

```diff
@@ -1,64 +1,116 @@
     
     protected static final Logger log = Logger.getLogger("datawave.ingest");
     
-    public TableConfigCache(Configuration conf) {
+    private TableConfigCache(Configuration conf) {
         super(conf);
     }
     
-    @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
-    public void setTableConfigs(Configuration conf) {
-        try {
-            TableConfigurationUtil tableConfig = new TableConfigurationUtil(conf);
-            configMap = tableConfig.getTableAggregatorConfigs();
-        } catch (Exception e) {
-            log.error("Unable to get table configurations " + e.getMessage());
+    public static TableConfigCache getCurrentCache(Configuration conf) {
+        if (null == cache) {
+            cache = new TableConfigCache(conf);
         }
+        return cache;
     }
     
-    @Override
-    public void update() {
-        setTableConfigs(this.conf);
-        super.update();
+    public void setTableConfigs(Map<String,Map<String,String>> confMap) {
+        configMap = confMap;
+    }
+    
+    public void clear() {
+        configMap = new HashMap();
+    }
+    
+    public boolean isInitialized() {
+        return !configMap.isEmpty();
     }
     
     @Override
     public void writeCacheFile(FileSystem fs, Path tmpCacheFile) {
         try (PrintStream out = new PrintStream(new BufferedOutputStream(fs.create(tmpCacheFile)), false, "UTF-8")) {
-            for (Map.Entry e : configMap.entrySet()) {
-                out.println(e.getKey() + DELIMITER + e.getValue());
+            for (Map.Entry<String,Map<String,String>> table : configMap.entrySet()) {
+                for (Map.Entry tableProp : table.getValue().entrySet()) {
+                    out.println(table.getKey() + DELIMITER + tableProp.getKey() + DELIMITER + tableProp.getValue());
+                }
             }
         } catch (IOException e) {
             log.error("Unable to write cache file " + tmpCacheFile, e);
         }
     }
     
+    @Override
+    protected void readCache(BufferedReader in, String delimiter) throws IOException {
+        this.configMap = new HashMap<>();
+        String line;
+        String table = null;
+        String propName;
+        String propVal;
+        Map<String,String> tempMap = new HashMap<>();
+        while ((line = in.readLine()) != null) {
+            String[] parts = StringUtils.split(line, delimiter);
+            if (table == null || !table.equals(parts[0])) {
+                table = parts[0];
+                tempMap = new HashMap<>();
+                this.configMap.put(table, tempMap);
+            }
+            if (parts.length == 3) {
+                propName = parts[1];
+                propVal = parts[2];
+                tempMap.put(propName, propVal);
+            } else {
+                throw new IOException("Invalid Table Config Cache at " + this.cacheFilePath + " . Please verify its contents.");
+            }
+        }
+        in.close();
```

Comment:
If an open stream is being passed in, then closing the stream is the responsibility of the caller.  No need to call in.close() here.